### ISSUE TITLE
'uninitialized constant Mime::HTML' was thrown after ActiveRecord::RecordNotFound

### ISSUE THREAD

**Issue by JevanWu** [rails/rails#29408]
I'm using `Rails 5.1.1` 

I expected to use `find_by!` to throw `ActiveRecord::RecordNotFound` to bring `404` out when user was reaching a unfound record.

But somehow, after the exception `ActiveRecord::RecordNotFound`, another error occurs which is `uninitialized constant Mime::HTML`. Finally, the `500` was thrown instead of `404`

Ruby version: `ruby 2.3.0p0`  Rails version: `Rails 5.1.1`

**Comment by alexcameron89** [comment]
Hey @JevanWu, can you provide us with a sample application that shows us the error you are having?

**Comment by JevanWu** [comment]
Hi @alexcameron89, the error has been resolved after upgrading the version of `web-console` from `2.3.0` to `3.5.1`. Thus, I've closed the issue. But I leave my error here for the people who encountering the same problem to search.

Thanks anyway!

The error I've got is 
```
ActiveRecord::RecordNotFound (Couldn't find User):

app/controllers/pages_controller.rb:6:in `home'

NameError: uninitialized constant Mime::HTML
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/web-console-2.3.0/lib/web_console/middleware.rb:58:in `acceptable_content_type?'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/web-console-2.3.0/lib/web_console/middleware.rb:36:in `block in call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/web-console-2.3.0/lib/web_console/middleware.rb:18:in `catch'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/web-console-2.3.0/lib/web_console/middleware.rb:18:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/actionpack-5.1.1/lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/railties-5.1.1/lib/rails/rack/logger.rb:36:in `call_app'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/railties-5.1.1/lib/rails/rack/logger.rb:24:in `block in call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/activesupport-5.1.1/lib/active_support/tagged_logging.rb:69:in `block in tagged'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/activesupport-5.1.1/lib/active_support/tagged_logging.rb:26:in `tagged'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/activesupport-5.1.1/lib/active_support/tagged_logging.rb:69:in `tagged'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/railties-5.1.1/lib/rails/rack/logger.rb:24:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/sprockets-rails-3.2.0/lib/sprockets/rails/quiet_assets.rb:13:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/actionpack-5.1.1/lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/actionpack-5.1.1/lib/action_dispatch/middleware/request_id.rb:25:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/rack-2.0.3/lib/rack/method_override.rb:22:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/rack-2.0.3/lib/rack/runtime.rb:22:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/activesupport-5.1.1/lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/actionpack-5.1.1/lib/action_dispatch/middleware/executor.rb:12:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/actionpack-5.1.1/lib/action_dispatch/middleware/static.rb:125:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/rack-2.0.3/lib/rack/sendfile.rb:111:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/railties-5.1.1/lib/rails/engine.rb:522:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/puma-3.6.2/lib/puma/configuration.rb:225:in `call'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/puma-3.6.2/lib/puma/server.rb:578:in `handle_request'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/puma-3.6.2/lib/puma/server.rb:415:in `process_client'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/puma-3.6.2/lib/puma/server.rb:275:in `block in run'
	from /Users/jevan/.rvm/gems/ruby-2.3.0/gems/puma-3.6.2/lib/puma/thread_pool.rb:116:in `block in spawn_thread'
```

**Comment by jeison24** [comment]
Hi, I have the same error.
I have the following code in my application_controller.rb
`if request.format == Mime::HTML`

and I got the following error:
*uninitialized constant Mime::HTML*

**Comment by ningzhang422** [comment]
Since Rails 5.0, `Mime::HTML` is removed. `Mime[:html]` return the same value.

**Comment by jeison24** [comment]
@ningzhang422 Thanks

**Comment by blindMoe** [comment]
Thank you very much for leaving this open!  It just saved me a lot of time!

**Comment by mateusluizfb** [comment]
In my case just updating "webmock" gem resolved the problem.

**Comment by siegy22** [comment]
I'm late to the party, but updating `web-console` solved this when I updated my rails 4.2 app to be a rails 5.2 app 😄
